dropshots/src/main/java/com/dropbox/dropshots/Dropshots.kt

```diff
@@ -79,48 +70,148 @@ public class Dropshots internal constructor(
+   * to disk to be pulled to the host machine to update the reference images.
+   *
+   * @param fileName lambda to create a filename from the snapshot name (i.e. the name provided when
+   * taking the snapshot), first parameter is the class name and the second is the test name.
+   * Defaults to `<class-name>_<func-name>.png` limited to 255 chars, replacing all spaces with underscores.
+   * @param filePath where the screenshots should be store in project under [com.dropbox.dropshots.DropshotsExtension.referenceOutputDirectory]
+   */
+  public fun assertSnapshot(
+    activity: Activity,
+    fileName: (String, String) -> String = defaultFilenameFunc,
+    filePath: String? = null,
+  ): Unit = assertSnapshot(Screenshot.capture(activity).bitmap, fileName, filePath)
+
+  /**
+   * Compares a screenshot of the visible screen content to a reference screenshot from the test application's assets.
+   *
+   * If `BuildConfig.IS_RECORD_SCREENSHOTS` is set to `true`, then the screenshot will simply be written
+   * to disk to be pulled to the host machine to update the reference images.
+   *
+   * @param fileName lambda to create a filename from the snapshot name (i.e. the name provided when
+   * taking the snapshot), first parameter is the class name and the second is the test name.
+   * Defaults to `<class name>_<func-name>.png` limited to 255 chars, replacing all spaces with underscores.
+   * @param filePath where the screenshots should be store in project under [com.dropbox.dropshots.DropshotsExtension.referenceOutputDirectory]
+   */
+  public fun assertSnapshot(
+    fileName: (String, String) -> String = defaultFilenameFunc,
+    filePath: String? = null,
+  ): Unit = assertSnapshot(Screenshot.capture().bitmap, fileName, filePath)
+
+
+  /**
+   * Compares a [bitmap] to a reference screenshot from the test application's assets.
+   *
+   * If `BuildConfig.IS_RECORD_SCREENSHOTS` is set to `true`, then the screenshot will simply be written
+   * to disk to be pulled to the host machine to update the reference images.
+   *
+   * @param bitmap the bitmap to assert
+   * @param fileName lambda to create a filename from the snapshot name (i.e. the name provided when
+   * taking the snapshot), first parameter is the class name and the second is the test name.
+   * Defaults to `<class name>_<func-name>.png` limited to 255 chars, replacing all spaces with underscores.
+   * @param filePath where the screenshots should be store in project under [com.dropbox.dropshots.DropshotsExtension.referenceOutputDirectory]
+   */
+  @Suppress("LongMethod")
+  public fun assertSnapshot(
+    bitmap: Bitmap,
+    fileName: (String, String) -> String = defaultFilenameFunc,
     filePath: String? = null,
   ) {
-    val filename = filenameFunc(name)
+    // Some CI filesystems don't support spaces in artifact names so we also have to replace them with underscores.
+    val filename = fileName(className, testName).replace(" ", "_")
```

Comment:
This underscore replacement means that callers **can't** alter that behavior to suite their needs. If you put it in the default filename lambda then we get the behavior in the default case, but users also have the flexibility to change that behavior if required. I'd recommend doing that, so that **all** filename generation happens within that function and we respect what it gives us.